### ISSUE TITLE
Orchestration Output and Exception Should contains a specific part about the exception.

### ISSUE THREAD

**Issue by bhugot** [Azure/azure-functions-durable-extension#2047]
For some use case It would be nice to be able to have more info in FunctionFailedException that would be in the Output in failed case  and not just a concatened string message

I see at least 2 more info :
ExceptionType
ExceptionMessage

This would allow some http status error handling in some case.

**Comment by cgillum** [comment]
@bhugot Just to make sure I'm understanding, would this be an alternative to using `FunctionFailedException.InnerException` to get this information?

**Comment by bhugot** [comment]
@cgillum first we should fix assignation as currently the exception.InnerException is assigned instead of exception
https://github.com/Azure/azure-functions-durable-extension/blob/636adc279601a5f4b38616eb572e987ad4377134/src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs#L727
Then we also need to have in in the output because if i'm right when you get orchestration status you don't have it

**Comment by cgillum** [comment]
I believe the code you highlighted is correct as-is because `e` is an instance of `TaskFailedException`, which itself is a wrapper exception used by the underlying Durable Task Framework. Therefore `e.InnerException` in this context represents the exception that was thrown by the app code, which is what you want to be the inner exception of `FunctionFailedException`.

**Comment by bhugot** [comment]
@cgillum yes it's indeed the inner exception but we don't have it in the output for now. I will make a proposal for this when I have a bit of time
